config/config.go

```diff
@@ -1,16 +1,70 @@
 package config
 
-import "time"
+import (
+	ldbopt "github.com/syndtr/goleveldb/leveldb/opt"
+)
 
-// ReplicaConfig specifies the information needed for a Replica and Version manager to
-// maintain global object versioning and provenance.
-// TODO: this configuration is pulled from a service context not a library context.
+// DefaultConfig is used if the user does not specify a configuration
+var DefaultConfig = Config{
+	Versions: ReplicaConfig{
+		PID:    1,
+		Region: "local",
+		Name:   "localhost",
+	},
+}
+
+// New creates a configuration with the required options and can also be used to specify
+// optional configuration e.g. for engine-specific operations.
+func New(options ...Option) Config {
+	// Create the default configuration in editable mode
+	conf := &Config{
+		Versions: ReplicaConfig{
+			PID:    DefaultConfig.Versions.PID,
+			Region: DefaultConfig.Versions.Region,
+			Name:   DefaultConfig.Versions.Name,
+		},
+	}
+
+	// Apply all options to the configuration
+	for _, opt := range options {
+		opt(conf)
+	}
+
+	// Return the value of the configuration
+	return *conf
+}
+
+// Config specifies the options necessary to open a Honu database.
+type Config struct {
+	Versions   ReplicaConfig
+	LDBOptions *ldbopt.Options
+}
```

Comment:
The idea is that we'll eventually populate this with `RocksDBOptions`, `PebbleDBOptions`, etc as well?

Reply:
Yes, I figured I'd leave it to the implementers of those engines since the story I had only dealt with leveldb.